packages/admin/admin/src/rte/XmlToState.ts

```diff
@@ -0,0 +1,194 @@
+import {
+    ContentState,
+    convertFromRaw,
+    DraftInlineStyleType,
+    EditorState,
+    RawDraftContentBlock,
+    RawDraftContentState,
+    RawDraftEntity,
+} from "draft-js";
+
+import { blockToXml } from "./stateToXml";
+
+interface InlineStyle {
+    id: number;
+    offset: number;
+    length: number;
+}
+
+interface EntityRange {
+    key: number;
+    offset: number;
+    length: number;
+}
+
+export const updateBlockContent = (block: RawDraftContentBlock) => {
+    const regexInlineStylesPattern = /<inline id="[0-9][0-9]?">|<\/inline>/g;
+    const regexEntitiesPattern = /<entity id="[0-9][0-9]?">|<\/entity>/g;
```

Comment:
I don't really understand what is happening here, but it looks like parsing XML/HTML with regex - which is a [bad idea](https://stackoverflow.com/a/1732454).

Reply:
I use this regex pattern to get all occurrences of the pseudo tags in the strings including their index. This is needed to update the inline styles and entity ranges correctly. I also use the regex to remove the pseudo tags to get a translation without the tags at the end. I know it's not a good solution, but I haven't found another way yet.